### PR TITLE
docs: correct small errors in yul.rst

### PR DESCRIPTION
**Pull request by minaminao**
- Change `first` -> `second`. `PUSH1 3 PUSH1 0x80 MLOAD ADD PUSH1 0x80 MSTORE` corresponds to the second line, not the first line.
- Clarify the description of `log0` data. It's clear from the instruction arguments that that's not true, but the description `log without topics and data mem[p...(p+s))` could be taken not to include `data`.
- Unify to `e.g.`

### CODE REVIEW

File: docs/yul.rst [hunk 1]
```diff
@@ -902,15 +902,15 @@ the ``dup`` and ``swap`` instructions as well as ``jump`` instructions, labels a
 +-------------------------+-----+---+-----------------------------------------------------------------+
 | invalid()               | `-` | F | end execution with invalid instruction                          |
 +-------------------------+-----+---+-----------------------------------------------------------------+
-| log0(p, s)              | `-` | F | log without topics and data mem[p...(p+s))                      |
+| log0(p, s)              | `-` | F | log data mem[p...(p+s)) without topics                          |
```
Comment (nikola-matic) [inline docs/yul.rst]: I'd omit the mention of topic here.

Comment (minaminao) [reply docs/yul.rst]: Thank you for reviewing! Omitted it.

**Review by NunoFilipeSantos — APPROVED** [review]
Thank you for your contribution!

File: docs/yul.rst [hunk 2]
```diff
@@ -152,7 +152,7 @@ where an object is expected.
 Inside a code block, the following elements can be used
 (see the later sections for more details):
 
-- literals, i.e. ``0x123``, ``42`` or ``"abc"`` (strings up to 32 characters)
+- literals, e.g. ``0x123``, ``42`` or ``"abc"`` (strings up to 32 characters)
```
Comment (cameel) [inline docs/yul.rst:155]: I think `e.g.` was actually more correct here. `i.e.` would suggest that these are all possible literals, but they're just examples.

**Review by cameel — COMMENTED** [review]
Thanks!

I think it needs a small tweak though.

File: docs/yul.rst (hunk 2 above)
Comment (cameel) [reply docs/yul.rst:155]: Ah, wait, my bad. That's exactly what you did here :) It's fine then.

**Review by cameel — APPROVED** [review]
Actually, everything's fine. Merging.

**Pull request merged**